### PR TITLE
Fix JsFuncHandler TemporalTrunc Century handling

### PR DESCRIPTION
**Pull request by drostron**
Fixes: #1966

### CODE REVIEW

File: frontend/src/test/scala/quasar/std/StdLibSpec.scala [hunk 1]
```diff
@@ -605,6 +605,19 @@ abstract class StdLibSpec extends Qspec {
...
+
+          truncZonedDateTime(y, t).fold(
+            e => Failure(e.shows),
+            tt => unary(
+              TemporalTrunc(y, _).embed,
+              Data.Timestamp(x),
+              Data.Timestamp(tt.toInstant)))
+        }
+
         "timestamp" >> prop { (x: Instant, y: TemporalPart) =>
```
Comment (sellout) [inline frontend/src/test/scala/quasar/std/StdLibSpec.scala]: Would probably be a good idea to explicitly property-test each of the values in `TemporalPart`.

Comment (drostron) [inline frontend/src/test/scala/quasar/std/StdLibSpec.scala]: Agreed.

**Comment by drostron** [comment]
Re recent discussion, I'm reminded that we might be able to leverage 47deg's scalacheck-datetime now that our scalacheck version has been updated. I'll give that a quick whirl while in the vicinity.

**Comment by drostron** [comment]
`scalacheck-datetime` doesn't have the distribution shaping I was looking for so leaving `DateArbitrary` as is for now.

**Comment by drostron** [comment]
@sellout Ready for another look.

**Pull request merged**